### PR TITLE
Switch to using imported module idents for marking version requirements

### PR DESCRIPTION
**Pull request by coolreader18**
# Description of Changes

Alternative to #233

# API and ABI

 - [x] This is a breaking change to the module ABI
 - [ ] This is a breaking change to the module API
 - [ ] This is a breaking change to the ClientAPI
 - [ ] This is a breaking change to the SDK API

*If the API is breaking, please state below what will break*

### CODE REVIEW

**Review by mamcx — APPROVED** [review]
LGTM

File: crates/cli/src/subcommands/generate/mod.rs [hunk 1]
```diff
@@ -374,7 +365,7 @@ fn extract_descriptions(wasm_file: &Path) -> anyhow::Result<ModuleDef> {
             }
         },
     )?;
-    linker.func_wrap("spacetime", "_buffer_alloc", WasmCtx::buffer_alloc)?;
+    linker.func_wrap("spacetime_6.0", "_buffer_alloc", WasmCtx::buffer_alloc)?;
```
Comment (RReverser) [inline crates/cli/src/subcommands/generate/mod.rs:368]: Nit: extract the constant somewhere at the top of the file?

File: crates/bindings-csharp/Runtime/bindings.c [hunk 2]
```diff
@@ -100,7 +100,11 @@ static MonoArray* stdb_buffer_consume(Buffer buf);
 //   return out;
 // }
 
-__attribute__((import_module("spacetime"),
+#define STDB_ABI_MAJOR_VERSION "6"
+#define STDB_IMPORT_MODULE_MINOR(minor) "spacetime_" STDB_ABI_MAJOR_VERSION "." #minor
+#define STDB_IMPORT_MODULE STDB_IMPORT_MODULE_MINOR(0)
```
Comment (RReverser) [inline crates/bindings-csharp/Runtime/bindings.c:105]: Nit: let's just inline the string here, same as Rust does. It's a bit weird to edit two parts of version in two places.

Comment (coolreader18) [reply crates/bindings-csharp/Runtime/bindings.c:105]: sure :+1:

File: crates/cli/src/subcommands/generate/mod.rs (hunk 1 above)
Comment (coolreader18) [reply crates/cli/src/subcommands/generate/mod.rs:368]: oh maybe I'll just use the constant from spacetimedb-lib

File: crates/bindings-csharp/Runtime/bindings.c (hunk 2 above)
Comment (RReverser) [reply crates/bindings-csharp/Runtime/bindings.c:104]: I meant inline minor into the same string too, so that it's just `spacetime_6.0` like in Rust.

**Pull request merged**